### PR TITLE
qt6: enable clang builds

### PR DESCRIPTION
**Pull request by ericLemanissier**
fixes conan-io/conan-center-index#21698

The original error is:
```
/home/conan/w/prod/BuildSingleReference/.conan/data/qt/6.3.0/_/_/source/qt6/qtbase/src/corelib/text/qlocale_tools.cpp:57:10: fatal error: 'charconv' file not found
#include <charconv>
         ^~~~~~~~~~
1 error generated.
```
It can be seen in https://c3i.jfrog.io/c3i/misc/logs/pr/10261/10-linux-clang/qt/6.3.0//8378bea5856cf76fce4bce4f3e960750c8a5402f-build.txt
see also my message https://github.com/conan-io/conan-center-index/pull/10261#issuecomment-1096869182

The original problem is that the docker images used by C3I for the build of clang11 profiles had a libstdc++ which is too old.
Now that clang 11 is not built by C3I any more, it may pass.

Specify library name and version:  **qt/6.***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
@stevenwdv @n-morales
I expect this to fail, because [C3I uses Xenial for its clang images](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L155) and the default version of libstdc++ on xenial is 6

**Comment by stevenwdv** [comment]
@ericLemanissier I had to look up what those terms meant but now I finally understand (although not why this version is used while a newer version is used for gcc). However, why should this impact my local build? I think the error should probably be removed because the conanfile has no idea what version of libstdc++ I have installed.

File: recipes/qt/6.x.x/conandata.yml [hunk 1]
```diff
@@ -67,14 +67,23 @@ patches:
     - "base_path": "qtwebengine"
       "patch_description": "Workaround for too long .rps file name"
       "patch_file": "patches/c72097e_6.6.0.diff"
```
Comment (MartinDelille) [inline recipes/qt/6.x.x/conandata.yml:69]: shouldn't we fix the linter issue ? it is polluting the review.

**Review by MartinDelille — APPROVED** [review]

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7329076005/attempts/2)

**Review by jwillikers — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**